cluster/validation.go

```diff
@@ -731,3 +737,19 @@ func getClusterVersion(version string) (semver.Version, error) {
 	}
 	return parsedVersion, nil
 }
+
+// warnWeaveDeprecation prints a deprecation warning if version is higher than 1.27 and weave was selected
+func warnWeaveDeprecation(k8sVersion, cniPlugin string) error {
+	version, err := util.StrToSemVer(k8sVersion)
+	if err != nil {
+		return fmt.Errorf("Error in the version: %v", version)
```

Comment:
Should probably reference `k8sVersion` (string to parse) instead of returned semver as I doubt it will return anything usable in case of error. Also would like to include/wrap the actual error returned.

```suggestion
		return fmt.Errorf("Error parsing K8s version '%s': %w", k8sVersion, err)
```

Reply:
good point